fili-core/src/main/java/com/yahoo/bard/webservice/table/PhysicalTable.java

```diff
@@ -137,14 +141,17 @@ public synchronized void resetColumns(SegmentMetadata segmentMetadata, Dimension
             workingIntervals.clear();
             for (Map.Entry<String, Set<Interval>> nameIntervals : dimensionIntervals.entrySet()) {
                 String physicalName = nameIntervals.getKey();
-                String apiName = getLogicalColumnName(physicalName);
-                Dimension dimension = dimensionDictionary.findByApiName(apiName);
-                // Schema evolution may lead to unknown dimensions, skip these
-                if (dimension == null) {
-                    continue;
+                List<String> apiNames = getLogicalColumnName(physicalName);
+
+                for (String apiName : apiNames) {
+                    Dimension dimension = dimensionDictionary.findByApiName(apiName);
+                    // Schema evolution may lead to unknown dimensions, skip these
+                    if (dimension == null) {
+                        continue;
+                    }
+                    DimensionColumn dimensionColumn = DimensionColumn.addNewDimensionColumn(this, dimension);
+                    workingIntervals.put(dimensionColumn, nameIntervals.getValue());
```

Comment:
You can streamify this easily enough, I think:

```java
 getLogicalColumnName(physicalName).stream()
     .map(dimensionDictionary::findByApiName)
     .filter(dimension -> dimension != null)
     .forEach(dimension -> {
           DimensionColumn dimensionColumn = DimensionColumn.addNewDimensionColumn(this, dimension);
           workingIntervals.put(dimensionColumn, nameIntervals.getValue()); 
     });
```

Note that you could put `DimensionColumn.addNewDimensionColumn(this, dimension)` in a `map` rather than the first statement of a `forEach`, but `addNewDimensionColumn` has side-effects, and side-effecting operations in a `map` operation feels ugly to me.